### PR TITLE
Feature/member custom notifications

### PR DESCRIPTION
**Pull request by ulferts**
Allow sending a custom message when:
* creating a membership
* updating a membership

As of now, it is not yet clear, whether sending a custom message can be sent to every user in a group in case a group's membership is created/updated.

## TODO

* [x] include `notificationMessage` in `payload/_meta` of membership form requests
* [x] accept `notificationMessage` in `_meta` on update/create requests to the membership resource 
* [x] include custom message in mails send to new/altered members
  * [x] override setting which is disabling sending of member updated/created mails in case a custom message is provided?
* [x] update API docs

https://community.openproject.org/projects/openproject/work_packages/36019

### CODE REVIEW

File: docs/api/apiv3/endpoints/members.apib [hunk 1]
```diff
@@ -203,6 +223,12 @@ Updates the given membership by applying the attributes provided in the body.
                         "href": "/api/v3/roles/8"
                       }
                     ]
+                },
+                "_meta": {
+                  "notificationMessage": {
+                    "raw" {
+                      "Alice, you can now also create work packages."
+                    }
```
Comment (oliverguenther) [inline docs/api/apiv3/endpoints/members.apib]:
```suggestion
                },
                "_meta": {
                  "notificationMessage": {
                     "raw": "Alice, you can now also create work packages."
                   }
```

This was not valid json

File: docs/api/apiv3/endpoints/members.apib [hunk 2]
```diff
@@ -525,17 +560,25 @@ For more details and all possible responses see the general specification of [Fo
...
+                      }
+                  ]
                 },
-                "roles": [
-                    {
-                        "href": "/api/v3/roles/2"
+                "_meta": {
+                  "notificationMessage": {
+                    "raw" {
+                      "Welcome to our team, Alice."
```
Comment (oliverguenther) [inline docs/api/apiv3/endpoints/members.apib]:
```suggestion
                "_meta": {
                  "notificationMessage": {
                    "raw": "Welcome to our team, Alice."
                    
```

File: docs/api/apiv3/endpoints/members.apib [hunk 3]
```diff
@@ -561,6 +604,12 @@ For more details and all possible responses see the general specification of [Fo
                                     "title": "Manager"
                                 }
                             ]
+                        },
+                        "_meta": {
+                          "notificationMessage": {
+                            "raw" {
+                              "Welcome to our team, Alice."
```
Comment (oliverguenther) [inline docs/api/apiv3/endpoints/members.apib]:
```suggestion
                            "raw": "Welcome to our team, Alice."
```

File: docs/api/apiv3/endpoints/members.apib [hunk 4]
```diff
@@ -686,18 +744,29 @@ The returned payload part of the form will be contain a merge between the alread
...
+                  "roles": [
+                    {
+                      "href": "/api/v3/roles/4"
+                    }
+                  ]
+                },
+                "_meta": {
+                  "notificationMessage": {
+                    "raw" {
+                      "Alice, you can now also create work packages."
```
Comment (oliverguenther) [inline docs/api/apiv3/endpoints/members.apib]:
```suggestion
                    "raw": "Alice, you can now also create work packages."
```

File: docs/api/apiv3/endpoints/members.apib [hunk 5]
```diff
@@ -715,6 +784,12 @@ For more details and all possible responses see the general specification of [Fo
                                     "title": "A role"
                                 }
                             ]
+                        },
+                        "_meta": {
+                          "notificationMessage": {
+                            "raw" {
+                              "Alice, you can now also create work packages."
```
Comment (oliverguenther) [inline docs/api/apiv3/endpoints/members.apib]:
```suggestion
                            "raw": "Alice, you can now also create work packages."
```

**Review by oliverguenther — CHANGES_REQUESTED** [review]
Works as expected :+1: I only have some minor docs remarks

File: docs/api/apiv3/endpoints/members.apib (hunk 1 above)
Comment (ulferts) [reply docs/api/apiv3/endpoints/members.apib]: My mind must have been blank at this point. Thanks.

**Review by oliverguenther — APPROVED** [review]

**Pull request merged**